src/test/java/com/github/mtakaki/credentialstorage/hibernate/UnitOfWorkApplicationListenerTest.java

```diff
@@ -0,0 +1,397 @@
+        inOrder.verify(this.sessionFactory).openSession();
+        inOrder.verify(this.session).close();
+    }
+
+    @Test
+    public void bindsAndUnbindsTheSessionToTheManagedContext() throws Exception {
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(final InvocationOnMock invocation) throws Throwable {
+                assertThat(ManagedSessionContext
+                        .hasBind(UnitOfWorkApplicationListenerTest.this.sessionFactory))
+                                .isTrue();
+                return null;
+            }
+        }).when(this.session).beginTransaction();
+
+        this.execute();
+
+        assertThat(ManagedSessionContext.hasBind(this.sessionFactory)).isFalse();
+    }
+
+    @Test
+    public void configuresTheSessionsReadOnlyDefault() throws Exception {
+        this.prepareAppEvent("methodWithReadOnlyAnnotation");
+
+        this.execute();
+
+        verify(this.session).setDefaultReadOnly(true);
+    }
+
+    @Test
+    public void configuresTheSessionsCacheMode() throws Exception {
+        this.prepareAppEvent("methodWithCacheModeIgnoreAnnotation");
+
+        this.execute();
+
+        verify(this.session).setCacheMode(CacheMode.IGNORE);
+    }
+
+    @Test
+    public void configuresTheSessionsFlushMode() throws Exception {
+        this.prepareAppEvent("methodWithFlushModeAlwaysAnnotation");
+
+        this.execute();
+
+        verify(this.session).setFlushMode(FlushMode.ALWAYS);
+    }
+
+    @Test
+    public void doesNotBeginATransactionIfNotTransactional() throws Exception {
+        final String resourceMethodName = "methodWithTransactionalFalseAnnotation";
+        this.prepareAppEvent(resourceMethodName);
+
+        when(this.session.getTransaction()).thenReturn(null);
+
+        this.execute();
+
+        verify(this.session, never()).beginTransaction();
+        verifyZeroInteractions(this.transaction);
+    }
+
+    @Test
+    public void detectsAnnotationOnHandlingMethod() throws NoSuchMethodException {
+        final String resourceMethodName = "handlingMethodAnnotated";
+        this.prepareAppEvent(resourceMethodName);
+
+        this.execute();
+
+        verify(this.session).setDefaultReadOnly(true);
+    }
+
+    @Test
+    public void detectsAnnotationOnDefinitionMethod() throws NoSuchMethodException {
+        final String resourceMethodName = "definitionMethodAnnotated";
+        this.prepareAppEvent(resourceMethodName);
+
+        this.execute();
+
+        verify(this.session).setDefaultReadOnly(true);
+    }
+
+    @Test
+    public void annotationOnDefinitionMethodOverridesHandlingMethod() throws NoSuchMethodException {
+        final String resourceMethodName = "bothMethodsAnnotated";
+        this.prepareAppEvent(resourceMethodName);
+
+        this.execute();
+
+        verify(this.session).setDefaultReadOnly(true);
+    }
+
+    @Test
+    public void beginsAndCommitsATransactionIfTransactional() throws Exception {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://www.codacy.com/app/mitsuotakaki/CredentialStorageService-dw-hibernate/file/2455877775/issues/source?bid=3237716&fileBranchId=3250434#l194)